src/bashDebug.ts

```diff
@@ -71,32 +77,73 @@ export class BashDebugSession extends LoggingDebugSession {
 			this.sendResponse(response)
 		});
 
-		spawn("bash", ["-c", `pkill -KILL -P ${this.debuggerProcessParentId}`]);
+		spawn("bash", ["-c", `${this.launchArgs.pathPkill} -KILL -P ${this.debuggerProcessParentId}`]);
+	}
+
+	/**
+	 * TODO: replace return type with something like errno in POSIX.
+	 */
+	private isPathValid(args: LaunchRequestArguments): boolean {
+		const argv = ["-c",
+			`type ${args.pathBashdb} || exit 1;` +
+			`type ${args.pathCat} || exit 2;` +
+			`type ${args.pathMkfifo} || exit 3;` +
+			`type ${args.pathPkill} || exit 4;` +
+			`test -d ${args.cwd} || exit 11;` +
+			""
+		]
+		const proc = spawnSync(args.pathBash, argv, { timeout: 1000 });
+
+		if (proc.error !== undefined) {
+			// @ts-ignore Property 'code' does not exist on type 'Error'.
+			if (proc.error.code === "ENOENT") {
+				// bash not found
+				return false;
+			}
+			// @ts-ignore Property 'code' does not exist on type 'Error'.
+			if (proc.error.code === "ETIMEDOUT") {
+				// TODO: It's possibly bug. Notify it to user.
+				return false;
+			}
+			// TODO: Else, unknown error. Show dialog to promote user to report the issue.
+			return false;
+		}
+
+		if (proc.status != 0) {
+			return false;
+		}
+
+		return true;
 	}
 
 	protected launchRequest(response: DebugProtocol.LaunchResponse, args: LaunchRequestArguments): void {
 
+		this.launchArgs = args;
+
 		// make sure to 'Stop' the buffered logging if 'trace' is not set
 		logger.setup(args.trace ? Logger.LogLevel.Verbose : Logger.LogLevel.Stop, false);
 
-		if (!args.bashDbPath) {
-			args.bashDbPath = "bashdb";
-		}
-
-		if (!args.bashPath) {
-			args.bashPath = "bash";
-		}
-
 		if (process.platform === "win32") {
 			args.cwd = `${expandPath(args.cwd)}`;
 			args.program = `${expandPath(args.program)}`;
 		}
 
+		if (!this.isPathValid(args)) {
+			// abort launch
```

Comment:
I think comments like those 2 can be ommited - it is quite obvious what code does.